massa-bootstrap/src/client_binder.rs

```diff
@@ -101,7 +129,30 @@ impl BootstrapClientBinder {
 
         // save prev sig
         self.prev_sig = Some(sig);
+        self.sended_message = None;
 
         Ok(message)
     }
+
+    #[allow(dead_code)]
+    /// Send a message to the bootstrap server
+    pub async fn send(&mut self, msg: BootstrapMessage) -> Result<(), BootstrapError> {
+        let msg_bytes = msg.to_bytes_compact()?;
+        let msg_len: u32 = msg_bytes.len().try_into().map_err(|e| {
+            BootstrapError::GeneralError(format!("bootstrap message too large to encode: {}", e))
+        })?;
+        if let Some(prev_sig) = self.prev_sig {
+            self.duplex.write_all(&prev_sig.to_bytes()).await?;
+        }
+        // send message length
+        {
+            let msg_len_bytes = msg_len.to_be_bytes_min(self.max_bootstrap_message_size)?;
+            self.duplex.write_all(&msg_len_bytes).await?;
+        }
```

Comment:
Why make a scope here? No need for a change I just don't understand the purpose.

Reply:
I just copied the behavior from the function to send on the server. I believe it's for split parts that are sent in the buffer and avoid a misusing of the variables after.